src/software/amazon/ionschema/internal/constraint/ContainerLength.kt

```diff
@@ -0,0 +1,36 @@
+package software.amazon.ionschema.internal.constraint
+
+import software.amazon.ion.IonContainer
+import software.amazon.ion.IonValue
+import software.amazon.ionschema.Violations
+import software.amazon.ionschema.Violation
+import software.amazon.ionschema.internal.CommonViolations
+import software.amazon.ionschema.internal.util.RangeFactory
+import software.amazon.ionschema.internal.util.RangeType
+
+/**
+ * Implements the container_length constraint.
+ *
+ * @see https://amzn.github.io/ion-schema/docs/spec.html#container_length
+ */
+internal class ContainerLength(
+        ion: IonValue
+) : ConstraintBase(ion) {
+
+    private val range = RangeFactory.rangeOf<Int>(ion, RangeType.INT_NON_NEGATIVE)
+
+    override fun validate(value: IonValue, issues: Violations) {
```

Comment:
The implementations for `byte_length`, `codepoint_length`, `container_length`, `scale`, and `precision` are all pretty similar. Seems like an opportunity for code re-use.